src/bptl/work_units/zgw/tasks/rol.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+from zgw_consumers.constants import APITypes
+
+from bptl.tasks.base import check_variable
+from bptl.tasks.registry import register
+
+from .base import ZGWWorkUnit
+
+logger = logging.getLogger(__name__)
+
+
+@register
+class CreateRolTask(ZGWWorkUnit):
+    """
+    Create a new ROL for the ZAAK in the process.
+
+    **Required process variables**
+
+    * ``zaakUrl``: full URL of the ZAAK to create a new rol for
+    * ``omschrijving``: roltype.omschrijving for the ROL
+    * ``betrokkeneType``: type of the betrokkene
+    * ``roltoelichting``: description of the ROL
+    * ``betrokkene``: JSON object containing the data for ROL.betrokkeneIdentificatie. See
+       https://zaken-api.vng.cloud/api/v1/schema/#operation/rol_create for the properties
+       available.
+    * ``services``: JSON Object of connection details for ZGW services:
+
+        .. code-block:: json
+
+          {
+              "<zrc alias>": {"jwt": "Bearer <JWT value>"},
+              "<ztc alias>": {"jwt": "Bearer <JWT value>"}
+          }
+
+    **Optional process variables**
+
+    * ``indicatieMachtiging``: Authorization indication ("gemachtigde" or "machtiginggever")
+
+    **Optional process variables (Camunda exclusive)**
+
+    * ``callbackUrl``: send an empty POST request to this URL to signal completion
+
+    **Sets the process variables**
+
+    * ``rolUrl``: the full URL of the created ROL
+    """
+
+    def create_rol(self) -> dict:
+        variables = self.task.get_variables()
+
+        zrc_client = self.get_client(APITypes.zrc)
+        zaak_url = check_variable(variables, "zaakUrl")
+        zaak = zrc_client.retrieve("zaak", url=zaak_url)
+
+        ztc_client = self.get_client(APITypes.ztc)
+        query_params = {
+            "zaaktype": zaak["zaaktype"],
+            "omschrijvingGeneriek": check_variable(variables, "omschrijving"),
```

Comment:
this is not what was specified in the ticket:

> `omschrijving`: the roltype omschrijving, use this in combination with zaak.zaaktype to find the correct rolType reference

`omschrijvingGeneriek` is not unique with regards to the zaaktype, but `omschrijving` is.

Reply:
according to catalogi schema (https://openzaak.utrechtproeftuin.nl/catalogi/api/v1/schema/#operation/roltype_list) `omschrijving` is not in the list of available query params, but `omschrijvingGeneriek` is in the list. Should I filter on it?

Reply:
no, it can't be queried. best is to just do the filtering in Python

Reply:
roltypen filtering on `omschrijving` is added